packages/jellyfish-api-core/__tests__/category/loan/paybackLoan.test.ts

```diff
@@ -1164,3 +1164,412 @@ describe('paybackLoan before FortCanningHeight', () => {
+      vaultId: vaultIdOneSat,
+      to: vaultOwnerAddress,
+      amounts: `${oneSat}@DUSD`
+    })
+    await testing.generate(1)
+
+    const vaultBefore = await testing.rpc.loan.getVault(vaultIdOneSat) as VaultActive
+    expect(vaultBefore.loanAmounts).toStrictEqual(['0.00000002@DUSD'])
+
+    await testing.rpc.loan.paybackLoan({
+      vaultId: vaultIdOneSat,
+      amounts: `${oneSat}@DFI`,
+      from: vaultOwnerAddress
+    })
+    await testing.generate(1)
+    const accAfterFirstPayback = await testing.rpc.account.getTokenBalances(undefined, undefined, { symbolLookup: true })
+    expect(accAfterFirstPayback).toContain('799999.99999999@DFI')
+
+    const burnInfoFirstPayback = await testing.rpc.account.getBurnInfo()
+    expect(burnInfoFirstPayback.paybackburn).toStrictEqual(new BigNumber(oneSat))
+    expect(burnInfoFirstPayback.dfipaybackfee).toStrictEqual(new BigNumber(0))
+
+    const vaultAfterFirstPayback = await testing.rpc.loan.getVault(vaultIdOneSat) as VaultActive
+    expect(vaultAfterFirstPayback.loanAmounts).toStrictEqual(['0.00000002@DUSD'])
+
+    await testing.rpc.loan.paybackLoan({
+      vaultId: vaultIdOneSat,
+      amounts: '0.00000005@DFI',
+      from: vaultOwnerAddress
+    })
+    await testing.generate(1)
+
+    const accAfterSecondPayback = await testing.rpc.account.getTokenBalances(undefined, undefined, { symbolLookup: true })
+    expect(accAfterSecondPayback).toContain('799999.99999996@DFI')
+
+    const burnInfoAfterSecondPayback = await testing.rpc.account.getBurnInfo()
+    expect(burnInfoAfterSecondPayback.paybackburn).toStrictEqual(new BigNumber(oneSat * 4))
+    expect(burnInfoAfterSecondPayback.dfipaybacktokens).toStrictEqual(['0.00000002@DUSD'])
+
+    const vaultAfterSecondPayback = await testing.rpc.loan.getVault(vaultIdOneSat) as VaultActive
+    expect(vaultAfterSecondPayback.loanAmounts).toHaveLength(0)
+  })
+
+  it('should be able to payback DUSD loan using DFI with excess DFI', async () => {
+    await testing.rpc.masternode.setGov({ [attributeKey]: { [key]: 'true' } })
+    await testing.generate(1)
+
+    const burnInfoBefore = await testing.rpc.account.getBurnInfo()
+    expect(burnInfoBefore.paybackburn).toStrictEqual(new BigNumber(0))
+    expect(burnInfoBefore.dfipaybackfee).toStrictEqual(new BigNumber(0))
+    expect(burnInfoBefore.dfipaybacktokens).toStrictEqual([])
+
+    const dusdInterestPerBlock = new BigNumber(netInterest * dusdLoanAmount / (365 * blocksPerDay))
+    const paybackLoanBlockHeight = await testing.rpc.blockchain.getBlockCount()
+    const dusdInterestAmountBefore = dusdInterestPerBlock.multipliedBy(new BigNumber(paybackLoanBlockHeight - dusdTakeLoanBlockHeight + 1))
+    const dusdLoanAmountBefore = new BigNumber(dusdLoanAmount).plus(dusdInterestAmountBefore.decimalPlaces(8, BigNumber.ROUND_CEIL))
+
+    // calculate how much dfi is required to pay off all dusd at a penalty rate of 1%
+    // dfi_needed = loan_amount/(1-(price*penalty_rate))
```

Comment:
This comment should be changed to "// dfi_needed = loan_amount/(price * (1 - penalty_rate))", right ?

Reply:
price as it depends on the oracle which happens to be set to one for this instance. to make it clearer im adding another comment to state how the price is taken if that is ok with you?